### PR TITLE
feat(backup): schedule expression timezone for backup plans

### PR DESCRIPTION
**Pull request by jogold**
Add timezone support for schedule expressions in backup plans.

See https://aws.amazon.com/about-aws/whats-new/2023/08/aws-backup-local-time-zone-selections/

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-backup/lib/rule.ts [hunk 1]
```diff
@@ -44,6 +44,17 @@ export interface BackupPlanRuleProps {
...
+  /**
+   * The name of the time zone for the schedule expression, following the IANA time zone standard.
+   *
+   * @example America/Los_Angeles
+   *
+   * @see https://www.iana.org/time-zones
+   *
+   * @default - UTC
+   */
+  readonly scheduleExpressionTimezone?: string;
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-backup/lib/rule.ts:56]: we export a `TimeZone` prop out of core.

File: packages/aws-cdk-lib/aws-backup/lib/rule.ts [hunk 2]
```diff
@@ -213,6 +224,10 @@ export class BackupPlanRule {
       throw new Error('`scheduleExpression` must be of type `cron`');
     }
 
+    if (props.scheduleExpressionTimezone && !props.scheduleExpression) {
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-backup/lib/rule.ts:227]: there are some `scheduleExpression`s that don't make sense with a timezone, and this check ignores that.

**Review by kaizencc — CHANGES_REQUESTED** [review]
Hi @jogold, I think we should implement this similar to what we do in `scheduler-alpha`:

https://github.com/aws/aws-cdk/blob/main/packages/@aws-cdk/aws-scheduler-alpha/lib/schedule-expression.ts

Of course, I still don't understand why we don't just invest time in getting the schedule class into core and I know you've tried exactly that before.

**Comment by jogold** [comment]
> Hi @jogold, I think we should implement this similar to what we do in `scheduler-alpha`:
> 
> [`main`/packages/@aws-cdk/aws-scheduler-alpha/lib/schedule-expression.ts](https://github.com/aws/aws-cdk/blob/main/packages/@aws-cdk/aws-scheduler-alpha/lib/schedule-expression.ts?rgh-link-date=2023-09-05T13%3A49%3A15Z)
> 
> Of course, I still don't understand why we don't just invest time in getting the schedule class into core and I know you've tried exactly that before.

OK, this means deprecating `scheduleExpression` in favor of `schedule`? `cronExpression`? other name? and you don't want to import the `ScheduleExpression` class of `aws-scheduler-alpha` in here?

File: packages/aws-cdk-lib/aws-backup/lib/rule.ts (hunk 1 above)
Comment (jogold) [reply packages/aws-cdk-lib/aws-backup/lib/rule.ts:56]: missed that one

**Comment by jogold** [comment]
> that `scheduleExpression` only accepts CRON expressions so I wonder what happens if you supply `events.Schedule.rate('1 minute')`.

https://github.com/aws/aws-cdk/blob/d69c51a2e10864e69d4610b6df30d40a9bfec3ac/packages/aws-cdk-lib/aws-backup/lib/rule.ts#L212-L214

> I hate this and want to fix it once and for all. What do you think @jogold? If you agree, I can spend some time trying to get the first part right and that should kick start the other changes.

Yeah, it seems to be the right thing to do.

**Comment by kaizencc** [comment]
Oh my that's a horrible interface in backup. What we're talking about seems like it will be an improvement across the board. Give me till the end of the week to stand up a PR to put abstract schedule in core?

**Comment by jogold** [comment]
Closing in favor of #27105

**Pull request closed without merging**